pipenv/project.py

```diff
@@ -610,12 +623,31 @@ def write_toml(self, data, path=None):
                         )
                         data[section][package].update(_data)
             formatted_data = toml.dumps(data).rstrip()
+        if os.path.abspath(path) == os.path.abspath(self.pipfile_location):
```

Comment:
Maybe use `os.path.samefile()` to handle case-insensitive filesystems better?

Reply:
`abspath` does not normalise paths, however. We should do `normpath(abspath(...))`.

Reply:
Good point